admin/sql/CreateViews.sql

```diff
@@ -83,6 +83,24 @@ CREATE OR REPLACE VIEW work_series AS
     LEFT OUTER JOIN link_attribute_text_value latv ON (latv.attribute_type = s.ordering_attribute AND latv.link = l.id)
     ORDER BY series, link_order;
 
+CREATE OR REPLACE VIEW recording_release_dates AS
+    SELECT track.recording,
+           release.id AS release,
+           date_year AS year,
+           date_month AS month,
+           date_day AS day
+      FROM track
+      JOIN medium ON (medium.id = track.medium)
+      JOIN release ON (release.id = medium.release)
+      LEFT JOIN (
+          SELECT release, date_year, date_month, date_day
+            FROM release_country
+           UNION
+          SELECT release, date_year, date_month, date_day
+            FROM release_unknown_country
+      ) release_dates
+        ON release_dates.release = release.id;
+
```

Comment:
Is having a view better than just doing the query in the function as with `set_release_group_first_release_date` then?

Reply:
The view helps a bit because there are actually two functions added:

 * `set_recordings_first_release_dates` - this updates the release dates for specific recording IDs (when a track is inserted/updated/deleted).
 * `set_releases_recordings_first_release_dates` - this updates the release dates for all recordings for specific release IDs (when a release event is inserted/updated/deleted).

Both of these functions require mostly identical queries except for the `recording = ?` and `release = ?` parts, so I put the common part of the query into a view.

I could have also implemented it where we just pass `set_recordings_first_release_dates` a giant list of recordings in the second case, but PG can't optimize that as well and I didn't want to overload these triggers.